pynest/nest/tests/test_NodeCollection.py

```diff
@@ -701,6 +701,55 @@ def test_array_indexing_bools(self):
             with self.assertRaises(err):
                 sliced = n[case]
 
+    def test_empty_nc(self):
+        """Connection with empty NodeCollection raises error"""
+        empty_nc = nest.NodeCollection([])
+        nodes = nest.Create('iaf_psc_alpha', 5)
+
+        with self.assertRaises(nest.kernel.NESTErrors.BadProperty):
+            nest.Connect(nodes, empty_nc)
+
+        with self.assertRaises(nest.kernel.NESTErrors.BadProperty):
+            nest.Connect(empty_nc, nodes)
+
+        with self.assertRaises(nest.kernel.NESTErrors.BadProperty):
+            nest.Connect(empty_nc, empty_nc)
+
+        with self.assertRaises(ValueError):
+            empty_nc.get()
+
+        with self.assertRaises(AttributeError):
+            empty_nc.V_m
+
+        self.assertFalse(empty_nc)
+        self.assertTrue(nodes)
+        self.assertIsNone(empty_nc.set())  # Also checking that it does not raise an error
+
+    def test_empty_nc_addition(self):
+        """Combine NodeCollection with empty NodeCollection and connect"""
+        n = 5
+        vm = -50.
+
+        nodes_a = nest.NodeCollection([])
+        nodes_a += nest.Create('iaf_psc_alpha', n)
+        nest.Connect(nodes_a, nodes_a)
+        self.assertEqual(nest.GetKernelStatus('num_connections'), n*n)
+        self.assertTrue(nodes_a)
+        self.assertIsNotNone(nodes_a.get())
+        nodes_a.V_m = vm
+        self.assertEqual(nodes_a.V_m, n * (vm,))
+
+        nest.ResetKernel()
+
+        nodes_b = nest.Create('iaf_psc_alpha', n)
+        nodes_b += nest.NodeCollection([])
```

Comment:
Could we have one of the tests use ``nest.NodeCollection()`` just to make sure it always works an no one removes it inadvertently?

Reply:
Done!